dbt/models/default/schema/default.vw_pin_value.yml

```diff
@@ -43,8 +49,48 @@ models:
         description: '{{ doc("shared_column_year") }}'
 
     tests:
+      - not_accepted_values:
+          name: default_vw_pin_value_mailed_class_no_hyphens
+          column_name: mailed_class
+          values: "2-99"
       - unique_combination_of_columns:
           name: default_vw_pin_value_unique_by_14_digit_pin_and_year
           combination_of_columns:
             - pin
             - year
+      - not_null:
+          name: default_vw_pin_value_mailed_class_not_null
+          column_name: mailed_class
+          config:
+            where: CAST(year AS int) < {{ var('test_qc_year_start') }}
+            error_if: ">289" # as of 2024-03-15
```

Comment:
I don't love this system of ignoring nulls. Is there are better way to handle them? Should we include them in the QC sheets? @jeancochrane do you have thoughts here?

Reply:
@dfsnow If the failure is caused by errors in an underlying iasWorld table, I do think it's worth adding a test to that table so that it gets added to the worksheet and fixed. It's a little bit harder to think about how to handle tests that are intended to test for incorrect logic in our views, but that can be led to fail based on issues in the underlying data; ideally we would be able to fully separate logic tests from data tests, probably by running our logic tests using a set of fixtures that populate tables with test data, but that's a much bigger lift and likely not possible in some cases. My first thought at an 80/20 solution would be to restrict the date range for the test to a time period where we know there are no errors, but we can discuss more during our infrastructure discussion if that option doesn't work.

Reply:
Alright, let's discuss more during Infrastructure Week.